libs/react-helpers/src/lib/grid/date-range-filter.tsx

```diff
@@ -1,113 +1,248 @@
+      ) {
+        setError(
+          t(
+            'The earliest data that can be queried is %s days ago.',
+            String(props.maxSubDays)
+          )
+        );
+        return false;
+      }
+      if (props?.maxDaysRange) {
+        const contrvalue =
+          name === 'start'
+            ? update?.end || value.end
+            : update?.start || value.start;
         if (
-          value.end &&
-          rowValue &&
-          new Date(rowValue) >= new Date(value.end)
+          Math.abs(
+            differenceInDays(new Date(timeValue), new Date(contrvalue))
+          ) > props.maxDaysRange
         ) {
+          setError(
+            t(
+              'The maximum time range that can be queried is %s days.',
+              String(props.maxDaysRange)
+            )
+          );
           return false;
         }
-        return true;
-      },
-
-      isFilterActive() {
-        return value.start || value.end;
-      },
-
-      getModel() {
-        if (!this.isFilterActive()) {
-          return null;
+      }
+      setError('');
+      return true;
+    };
+    const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+      const { value: dateValue, name } = event.target;
+      const date = new Date(dateValue || defaultDates[name as 'start' | 'end']);
+      const stringedDate = isValid(date) ? formatRFC3339(date) : undefined;
+      let update = { [name]: stringedDate };
+      if (name === 'start' && props.maxDaysRange) {
+        const endDate = new Date(value.end || Date.now());
+        if (Math.abs(differenceInDays(date, endDate)) > props.maxDaysRange) {
+          update = {
+            ...update,
+            end: formatRFC3339(
+              min([new Date(), addDays(date, props.maxDaysRange)])
+            ),
+          };
+        } else if (isBefore(endDate, date)) {
+          update = {
+            ...update,
+            end: formatRFC3339(min([new Date(), addDays(date, 1)])),
+          };
         }
-
-        return { value };
-      },
-
-      setModel(model?: { value: Schema.DateRange } | null) {
-        setValue(model?.value || defaultFilterValue);
-      },
+      }
+      if (validate(name, date, update)) {
+        setValue({
+          ...value,
+          ...update,
+        });
+      }
     };
-  });
+    useEffect(() => {
+      props?.filterChangedCallback();
+    }, [value, props]);
 
-  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue({
-      ...value,
-      [event.target.name]:
-        event.target.value &&
-        new Date(event.target.value).toISOString().replace('Z', '000000Z'),
-    });
-  };
-
-  useEffect(() => {
-    props?.filterChangedCallback();
-  }, [value, props]);
+    const Notification = () => {
+      const not = error ? (
+        <div className="border-vega-pink bg-vega-pink-300 dark:bg-vega-pink-650 border rounded p-2 flex items-start gap-2.5 m-4 w-full">
```

Comment:
chore(blocking): Use InputError underneath the date inputs

Reply:
can't easily do this because there are circular dependencies between `react-helpers` and `ui-toolkit`

Reply:
Moreover we need rather to put errors above, because if native date picker is expanded, it cover everything what is under - that's why I left some space over inputs

Reply:
Yikes, we'll need to fix that. Okay makes sense to go above, but can we just make it red text so it looks the same as InputError